src/scripts/setup_ezdeploy.sh

```diff
@@ -16,197 +17,166 @@ error_log() {
-      ;;
-    3) export mlz_tier2_subid=${OPTARG}
-      add_unique_sub_to_array "${OPTARG}"
-      ;;
-    ?)
-      echo "Invalid option: -${OPTARG}."
-      exit 2
-      ;;
-  esac
+# check for dependencies
+"${this_script_path}/util/checkforazcli.sh"
+"${this_script_path}/util/checkfordocker.sh"
+
+# check docker strategy
+if [[ $docker_strategy != "local" && \
+      $docker_strategy != "build" && \
+      $docker_strategy != "load" ]]; then
+  error_log "ERROR: Unrecognized docker strategy detected. Must be 'local', 'build', or 'load'."
+  exit 1
+fi
+
+# create MLZ configuration file based on user input
+mlz_config_file="${src_path}/mlz_tf_cfg.var"
+echo "INFO: creating a MLZ config file based on user input at $(realpath "$mlz_config_file")..."
+
+### derive args from user input
+gen_config_args=()
+gen_config_args+=("-f ${mlz_config_file}")
+gen_config_args+=("-e ${tf_environment}")
+gen_config_args+=("-m ${metadata_host}")
+gen_config_args+=("-z ${mlz_env_name}")
+gen_config_args+=("-l ${mlz_config_location}")
+gen_config_args+=("-s ${mlz_config_subid}")
+gen_config_args+=("-t ${mlz_tenantid}")
+
+### add hubs and spokes, if present
+for j in "${subs_args[@]}"
+do
+  gen_config_args+=("$j")
 done
 
+### expand array into a string of space separated arguments
+gen_config_args_str=$(printf '%s ' "${gen_config_args[*]}")
+
+### create the file
+### do not quote args $gen_config_args_str, we intend to split
+### ignoring shellcheck for word splitting because that is the desired behavior
+# shellcheck disable=SC2086
+"${this_script_path}/config/generate_config_file.sh" $gen_config_args_str
+
 # generate MLZ configuration names
-. "${BASH_SOURCE%/*}/config/generate_names.sh"  "bypass"
-
-# create the subscription resources
-. "${BASH_SOURCE%/*}/config/mlz_config_create.sh"  "bypass"
-
-for sub in "${subs[@]}"
-  do
-  echo "Setting Contributor role assignment for ${mlz_sp_name} on subscription ID: ${sub}"
-  az role assignment create \
-      --role Contributor \
-      --assignee-object-id "${sp_objid}" \
-      --scope "/subscriptions/${sub}" \
-      --assignee-principal-type ServicePrincipal \
-      --output none
-  done
-
-echo "INFO: Setting current az cli subscription to ${mlz_config_subid}"
-az account set --subscription "${mlz_config_subid}"
+. "${this_script_path}/config/generate_names.sh" "$mlz_config_file"
 
-# Handle Deployment of Login Services
+# create MLZ required resources
+echo "INFO: setting up required MLZ resources using $(realpath "$mlz_config_file")..."
+"${this_script_path}/config/mlz_config_create.sh" "$mlz_config_file"
 
-# Handle Remote Deploy to a Container Instance
-if [[ $docker_strategy != "local" ]]; then
-  echo "Creating ACR"
-  az acr create \
-  --resource-group "${mlz_rg_name}" \
-  --name "${mlz_acr_name}" \
-  --sku Basic
+# switch to the MLZ subscription
+echo "INFO: setting current az cli subscription to ${mlz_config_subid}..."
+az account set --subscription "${mlz_config_subid}"
 
```

Comment:
Why don't we take this one step further and get the rest of the way to @brooke-hamilton's ask of letting them just send a subscription?   We can retrieve the tenant for the sub by adding: 

 az account show --query "tenantId" -o tsv

Reply:
Great idea. Will update 👍